include/chainbase/undo_index.hpp

```diff
@@ -521,7 +521,7 @@ namespace chainbase {
          if( _undo_stack.size() != 0 )
             BOOST_THROW_EXCEPTION( std::logic_error("cannot set revision while there is an existing undo stack") );
 
-         if( revision > std::numeric_limits<int64_t>::max() )
+         if( revision > static_cast<uint64_t>(std::numeric_limits<int64_t>::max()) )
```

Comment:
Thanks @heifner for the background information! I was afraid to change `_revision` to `uint64_t`. 

I will do the right fix in this PR. This will prevent future confusions, remove all the related static_cast and make code cleaner.